diff --git a/src/backend/dab_decoder.h b/src/backend/dab_decoder.h
index 85f86af..50e7a2c 100644
--- a/src/backend/dab_decoder.h
+++ b/src/backend/dab_decoder.h
@@ -27,7 +27,10 @@
 #include <string>
 #include <vector>
 
+#if _FILE_OFFSET_BITS!=64
 #define MPG123_NO_LARGENAME	// disable large file API here
+#endif
+
 #include "mpg123.h"
 #if MPG123_API_VERSION < 36
 #	error "At least version 1.14.0 (API version 36) of mpg123 is required!"
