diff -ru librtlsdr-0.5.4.orig/include/tuner_r82xx.h librtlsdr-0.5.4.patched/include/tuner_r82xx.h
--- librtlsdr-0.5.4.orig/include/tuner_r82xx.h	2018-04-29 00:30:45.000000000 +0200
+++ librtlsdr-0.5.4.patched/include/tuner_r82xx.h	2018-08-10 15:04:53.000000000 +0200
@@ -115,6 +115,7 @@
 int r82xx_init(struct r82xx_priv *priv);
 int r82xx_set_freq(struct r82xx_priv *priv, uint32_t freq);
 int r82xx_set_gain(struct r82xx_priv *priv, int set_manual_gain, int gain);
+int r82xx_read_gain(struct r82xx_priv *priv);
 int r82xx_set_bandwidth(struct r82xx_priv *priv, int bandwidth,  uint32_t rate);
 
 #endif
diff -ru librtlsdr-0.5.4.orig/src/librtlsdr.c librtlsdr-0.5.4.patched/src/librtlsdr.c
--- librtlsdr-0.5.4.orig/src/librtlsdr.c	2018-04-29 00:30:45.000000000 +0200
+++ librtlsdr-0.5.4.patched/src/librtlsdr.c	2018-08-10 15:04:53.000000000 +0200
@@ -258,6 +258,12 @@
 	rtlsdr_dev_t* devt = (rtlsdr_dev_t*)dev;
 	return r82xx_set_gain(&devt->r82xx_p, 1, gain);
 }
+
+int r820t_read_gain(void *dev) {
+	rtlsdr_dev_t* devt = (rtlsdr_dev_t*)dev;
+	return r82xx_read_gain(&devt->r82xx_p);
+}
+
 int r820t_set_gain_mode(void *dev, int manual) {
 	rtlsdr_dev_t* devt = (rtlsdr_dev_t*)dev;
 	return r82xx_set_gain(&devt->r82xx_p, manual, 0);
@@ -1054,6 +1060,14 @@
 	if (!dev)
 		return 0;
 
+	if (dev->tuner_type == RTLSDR_TUNER_R820T) {
+		int gain;
+		rtlsdr_set_i2c_repeater(dev, 1);
+		gain = r820t_read_gain((void *)dev); // return the RF indicator
+		rtlsdr_set_i2c_repeater(dev, 0);
+		return gain;
+	}
+
 	return dev->gain;
 }
 
diff -ru librtlsdr-0.5.4.orig/src/tuner_r82xx.c librtlsdr-0.5.4.patched/src/tuner_r82xx.c
--- librtlsdr-0.5.4.orig/src/tuner_r82xx.c	2018-04-29 00:30:45.000000000 +0200
+++ librtlsdr-0.5.4.patched/src/tuner_r82xx.c	2018-08-10 15:04:53.000000000 +0200
@@ -974,16 +974,27 @@
 	return 0;
 }
 
-static int r82xx_read_gain(struct r82xx_priv *priv)
+int r82xx_read_gain(struct r82xx_priv *priv)
 {
+	int rf_indicator = 0;
 	uint8_t data[4];
 	int rc;
 
 	rc = r82xx_read(priv, 0x00, data, sizeof(data));
+
 	if (rc < 0)
 		return rc;
 
-	return ((data[3] & 0x0f) << 1) + ((data[3] & 0xf0) >> 4);
+	rf_indicator = ((10*(255-data[2]))/4);
+
+	printf("r82xx_read_gain %p rc=%d data[3]=0x%02x/%d rf_indicator=%d regs=(%02x.%02x.%02x.%02x)\n",
+		priv,
+		rc,
+		data[2],data[2],
+		rf_indicator,
+		data[0],data[1],data[2],data[3]);
+
+	return rf_indicator;
 }
 
 /* measured with a Racal 6103E GSM test set at 928 MHz with -60 dBm
@@ -1064,8 +1075,12 @@
 		if (rc < 0)
 			return rc;
 
+#if 1
+		rc = r82xx_write_reg_mask(priv, 0x0c, (1<<6)|(1<<4)|0xb, 0x5f);
+#else
 		/* set fixed VGA gain for now (26.5 dB) */
 		rc = r82xx_write_reg_mask(priv, 0x0c, 0x0b, 0x9f);
+#endif
 		if (rc < 0)
 			return rc;
 	}
